src/computation.ts

```diff
@@ -34,19 +34,17 @@ export const op = <C, A = Result<C>>(env: Env<C, A>): Computation<Env<C, A>, A,
   *[Symbol.iterator](): Iterator<Env<C, A>, A, A> { return yield env }
 }) as Computation<Env<C, A>, A, A>
 
-export const runComputation = <Y extends Env<any, N>, R, N> (g: Computation<Y, R, N>): Env<Capabilities<Y>, R> => {
-  const i = startComputation(g)
+export const runComputation = <Y extends Env<any, N>, R, N>(g: Computation<Y, R, N>): Env<Capabilities<Y>, R> =>
+  chainEnv(pureEnv(g), startComputation)
+
+const startComputation = <Y extends Env<any, N>, R, N>(g: Computation<Y, R, N>): Env<Capabilities<Y>, R> => {
```

Comment:
I think ideally this would be something like
```typescript
const startComputation = <Y extends Env<any, N>, R, N>(g: Computation<Y, R, N>): Env<Capabilities<Y>, R> => 
  chainEnv(i => stepComputation(i, i.next()), _ => resumeNow(g[Symbol.iterator]()))
```

Reply:
Hmmm, can you say more about the advantage of that, given that runComputation above already wraps g in Pure and uses chainEnv?

Reply:
I don't think there is one. I'm getting things confused with how I've represented `Env` in the past with how the implementation here is a function. Ignore me 👍

Reply:
👍 No worries!